src/codesyncd/websocket/socket_events.ts

```diff
@@ -115,6 +149,22 @@ export class SocketEvents {
         setDiffsBeingProcessed(diffsBeingProcessed);
     }
 
+    async onTabProcessed(tabFileName: string) {
+        const tabHandler = new TabHandler();
```

Comment:
Should add 
```if (!this.canSendSocketData) return;```
Keeping in view `onSyncSuccess`

Reply:
Good suggestion, thanks